src/distrs/binom.jl

```diff
@@ -16,4 +16,13 @@ import .RFunctions:
 binompdf(n::Real, p::Real, k::Real) = exp(binomlogpdf(n, p, k))
 
 # logpdf for numbers with generic types
-binomlogpdf(n::Real, p::Real, k::Real) = -log1p(n) - logbeta(n - k + 1, k + 1) + k * log(p) + (n - k) * log1p(-p)
+function binomlogpdf(n::Real, p::Real, k::Real)
+    if isinteger(k) & (zero(k) <= k <= n)
+        x = loggamma(n + 1) - loggamma(k + 1) - loggamma(n - k + 1) +
+            xlogy(k, p) + xlogy(n - k, 1 - p)
+    else
+        x = -Inf
+    end
+
+    return convert(typeof(float(p)), x)
```

Comment:
Actually this conversion seems to create problems with dual numbers, as shown by the CI failure. One solution would probably be to do `oftype(float(promote(n, p, k)), x)` (haven't checked).

Reply:
this is getting outside my area of expertise, I can make your suggested change if you want, but I can't vouch for its correctness.

Reply:
At least if tests pass that would be a good sign...

Reply:
ok, just made the change